compiler/lib/generate.ml

```diff
@@ -1124,8 +1124,7 @@ let rec translate_expr ctx queue loc x e level : _ * J.statement_list =
   | Special (Alias_prim name) ->
       let prim = Share.get_prim (runtime_fun ctx) name ctx.Ctx.share in
       (prim, const_p, queue), []
-  | Special Undefined ->
-      (J.(EVar (ident (Utf8_string.of_string_exn "undefined"))), const_p, queue), []
+  | Special Undefined -> (zero, const_p, queue), []
```

Comment:
Maybe rename the Undefined constructor.

Reply:
I have now just removed `Special Undefined` and use the integer constant `0` from the start.